src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        writeDataToFile(getDataFromFile(fromFileName), toFileName);
+    }
+
+    private static Map<String, Integer> getDataFromFile(String fromFileName) {
```

Comment:
why static?

Reply:
I made this method more flexible to read any string (not only buy or supply), but answering your question I didn't have any reasons to make it static (already fixed). I refactored code, made it simple as possible to execute only required task.